core/templates/dev/head/pages/Base.js

```diff
@@ -28,6 +29,13 @@ oppia.controller('Base', [
     $scope.isSidebarShown = SidebarStatusService.isSidebarShown;
     $scope.closeSidebarOnSwipe = SidebarStatusService.closeSidebar;
 
+    // Listner to catch the change in language prefrence
+    var listener = function(evt, translationResponse) {
```

Comment:
`listener` is a vague name, especially now that it's part of a larger piece of code. How about doing something like this?

```
$rootScope.$on('$translateChangeSuccess', function(evt, translationResponse) {
  ...
}));
```